src/y0/graph.py

```diff
@@ -518,10 +515,9 @@ def get_markov_blanket(self, nodes: Union[Variable, Iterable[Variable]]) -> Set[
             nodes = set(nodes)
         blanket = set()
         for node in nodes:
+            blanket.update(self.get_district(node))
```

Comment:
@pnavada With such a change, it's crucial to add tests that show how the current implementation did not work. This is also the perfect opportunity to write a detailed commit message about _why_ the current implementation didn't work. As it is, this can't be reviewed

Reply:
Yes, I agree that this would be a perfect oppotunity to write a detailed commit message.

I just referred the ananke implementation and replicated the same in y0.
The previous implementation defines markov blanket as the union of parents, children and parents of children of a given node.
But the ananke implementation behaves differently taking the union of the district and parents.
Most of the test cases defined failed for the previous implementation. And all of them pass for the current

Pasting the ananke code below for your reference:

```
def markov_blanket(self, vertices):
        """
        Get the Markov blanket of a set of vertices.

        :param vertices: iterable of vertex names.
        :return: set corresponding to Markov blanket.
        """

        blanket = set()
        for v in vertices:
            blanket = blanket.union(self.district(v))
        blanket = blanket.union(self.parents(blanket))
        return blanket - set(vertices)
```

add tests that show how the current implementation did not work - Let me know what exactly needs to be done here. I will work on it.